lib/faker/types.rb

```diff
@@ -0,0 +1,100 @@
+module Faker
+  class Types < Base
+    CHARACTERS = ('0'..'9').to_a + ('a'..'z').to_a
+    SIMPLE_TYPES = ["string", "fixnum"]
+    COMPLEX_TYPES = ["hash", "array"]
+    
+    class << self
+      def string(words=1)
+        resolved_num = resolve(words)
+        word_list = (
+          translate('faker.lorem.words')
+        )
+        word_list = word_list * ((resolved_num / word_list.length) + 1)
+        shuffle(word_list)[0, resolved_num].join(" ")
+      end
+      
+      def character
+        sample(CHARACTERS)
+      end
+      
+      def integer(from=0, to=100)
+        rand(from..to).to_i
+      end
+      
+      def hash(key_count=1)
+        hsh = {}
+        key_count.times do
+          hsh = hsh.merge({self.string.to_sym => self.random_type})
+        end
+        hsh
+      end
+      
+      def complex_hash(key_count=1)
+        hsh = {}
+        key_count.times do
+          hsh = hsh.merge({self.string.to_sym => self.random_complex_type})
+        end
+        hsh
+      end
+      
+      def array(len=1)
+        ar = []
+        len.times do
+          ar.push self.random_type
+        end
+        ar
+      end
+      
+      def constant(mod, key, val)
+        key = self.string if key.nil?
+        val = self.string if val.nil?
+        mod.const_set(titleize(key), val) if mod
+        Object.const_set(titleize(key), val) unless mod # sets const on main object, be CAREFUL!!!
+      end
+      
+      def random_type
```

Comment:
Consider doing something like this instead

```
DEFAULT_TYPE = :integer
TYPE_MAPPING = {
  string: self.string
  fixnum: self.integer
  hash: self.hash
  array: self.array
}

def get_random_type(type, types=SIMPLE_TYPES, type_mapping=TYPE_MAPPING, default_type=DEFAULT_TYPE)
type_mapping[type] || default_type


get_random_complex_type(type, types=COMPLEX_TYPES, type_mapping=TYPE_MAPPING, default_type=DEFAULT_TYPE)
  get_random_type(type, types, type_mapping, default_type)
```

This keeps your code DRY and adheres more closely to the Open/Closed principle. Your method is open to extension, but closed to modification. Now anyone that wants to subclass or add to the cases you outline can do so by updating the arguments being passed in instead of modifying the method logic.